File: backend/app/dtos/dice.py
from typing import List, Optional
from pydantic import BaseModel, Field, TypeAdapter

from app.dtos.base import BaseDTO
from app.dtos.players import PlayerDTO


class DiceDTO(BaseDTO):
    dice: str = Field(default="")
    locked: bool = Field(default=False)


class DiceShowDTO(BaseDTO):
    dice: int = Field(default=0)
    locked: bool = Field(default=False)
    show: str = Field(default="")


class UserBulletsDTO(BaseDTO):
    user_name: str = Field(default="")
    shots: int = Field(default=0)


class ShotsDTO(BaseModel):
    user_name: str = Field("Nome de quem vai receber os tiros")
    shots: int = Field("Quantidade de tiros que você vai dar nesse jogador")


class ListShotsDTO(BaseModel):
    shots_list: list[ShotsDTO] = Field("lista de shots")


shorts_list_adapter = TypeAdapter(List[ShotsDTO])


class ExecuteDistanceDTO(BaseDTO):
    bullet_total: int = Field(default=0)
    players_options: list[PlayerDTO] = Field(default_factory=list)
    user_bullets: list[UserBulletsDTO] = Field(default_factory=list)


class ExecuteDicesDTO(BaseDTO):
    current_player: PlayerDTO
    current_identity: str = Field(default="")
    one_distance: ExecuteDistanceDTO
    two_distance: Optional[ExecuteDistanceDTO] = Field(default=None)

File: mobile-gun-firing/src/dtos/dice.ts
import { userBullets } from "../components/shot";
import { Player } from "./players";

export type Dice = 1 | 2 | 3 | 4 | 5 | 6;

export type DiceOptions =
    | "1"
    | "2"
    | "Cerveja"
    | "Dinamite"
    | "Metralhadora"
    | "Flexa";

export type DiceCombination = {
    dice: Dice;
    show: DiceOptions;
    locked: boolean;
};

export type DiceCombinationUndefined = DiceCombination | undefined;

export const DICES: DiceCombination[] = [
    { dice: 1, show: "1", locked: false },
    { dice: 2, show: "2", locked: false },
    { dice: 3, show: "Cerveja", locked: false },
    { dice: 4, show: "Dinamite", locked: false },
    { dice: 5, show: "Metralhadora", locked: false },
    { dice: 6, show: "Flexa", locked: false },
];

export type ExecuteDistanceDTO = {
    bullet_total: number;
    players_options: Player[];
    user_bullets: userBullets[];
};

export type ExecuteDicesDTO = {
    current_player: Player;
    current_identity: string;
    game_id: string;
    table_situation: string;
    one_distance: ExecuteDistanceDTO;
    two_distance?: ExecuteDistanceDTO;
};
